### PR TITLE
feat: introduced session timeout check

### LINKED ISSUE #28: [Feat Request] Introduce session check when running spec
### Is your feature request related to a problem? Please describe
When running tests, there's a good change that we either forget to log in or that a session that we obtained earlier has expired.
If this is the case, we are blocked on the login page and the tool performs a series of retries that will lead to an unclear error after timing out.

### Describe the solution you'd like
I'm suggesting a solution that I've implemented in the E-Bikes sample app.

Let's save a timestamp when we generate the env file with the login URL:
https://github.com/trailheadapps/ebikes-lwc/blob/main/scripts/generate-login-url.js#L36

In our spec, we then read the timestamp and figure out if if the session has expired:
https://github.com/trailheadapps/ebikes-lwc/blob/main/force-app/test/utam/page-explorer.spec.js#L27

If this is the case, we abort the test with a clear error message.
https://github.com/trailheadapps/ebikes-lwc/blob/main/force-app/test/utam/page-explorer.spec.js#L31

### PR DESCRIPTION
**Pull request by pozil**
This PR addresses #28.

It introduces a new `SCRATCH_ORG_LOGIN_TIMESTAMP` env parameter that tracks session timeout.
When building the Salesforce environment during tests, we'll check for session timeout and halt the process with a clear error message if the session has timed out.

Closes #28

### CODE REVIEW

**Comment by olivier-martin-sf** [comment]
Thank you for your contribution @pozil, hopefully we will be able to review before end of the week and proceed with merging the changes

File: force-app/test/utilities/test-environment.js [hunk 1]
```diff
@@ -28,6 +28,9 @@
  * When creating an instance of this class, pass the environment name prefix as a parameter to the constructor
  * (i.e new TestEnvironment("sandbox"))
  */
+
+const SESSION_TIMEOUT = 2 * 60 * 1000; // 2 hours by default
```
Comment (trevor-bliss) [inline force-app/test/utilities/test-environment.js:32]: Is this the default timeout for scratch orgs created with SFDX CLI?

**Review by trevor-bliss — APPROVED** [review]
I think the changes make sense. I did not run this locally myself to verify functionality though FYI.

File: force-app/test/utilities/test-environment.js [hunk 2]
```diff
@@ -47,6 +50,13 @@ export class TestEnvironment {
         this.#username = this.#getKeyFromEnv('username');
         this.#password = this.#getKeyFromEnv('password');
         this.#sfdxLoginUrl = this.#getKeyFromEnv('loginUrl');
+
+        // Halt process if Salesforce session timed out
+        const loginTimestamp = this.#getKeyFromEnv('loginTimestamp');
+        if (!loginTimestamp || new Date().getTime() - parseInt(loginTimestamp, 10) > SESSION_TIMEOUT) {
+            console.error('FATAL ERROR: Salesforce session timed out. Re-authenticate before running tests.');
```
Comment (olivier-martin-sf) [inline force-app/test/utilities/test-environment.js:57]: nit: I would use something a bit less scary than `FATAL ERROR` for the error label as we aren't in a critical context (i.e kernel)

**Review by olivier-martin-sf — APPROVED** [review]
LGTM as well! Congrats for your contribution @pozil and many thanks from the team

File: force-app/test/utilities/test-environment.js (hunk 1 above)
Comment (pozil) [reply force-app/test/utilities/test-environment.js:32]: Yes, this is the default timeout.

Screen Shot 2022-05-21 at 08 35 06

File: force-app/test/utilities/test-environment.js (hunk 2 above)
Comment (pozil) [reply force-app/test/utilities/test-environment.js]: Ok, downgrading this to just ERROR.

**Comment by pozil** [comment]
Thanks for the review, happy to help.

**Pull request merged**